Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("MultiplayerService"));
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::OnTick(float, AZ::ScriptTimePoint)
+    {
+        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
+        if (!viewport)
+        {
+            return;
+        }
+
+        // Display the custom center viewport text
+        if (!m_centerViewportDebugText.empty())
+        {            
+            AzFramework::WindowSize viewportSize = viewport->GetViewportSize();
+            const float dpiScalingFactor = viewport->GetDpiScalingFactor();
+            const float center_screenposition_x = 0.5f*viewportSize.m_width;
+            const float center_screenposition_y = 0.5f*viewportSize.m_height;
+            const float screenposition_title_y = center_screenposition_y-9;
+            const float screenposition_debugtext_y = screenposition_title_y+18*dpiScalingFactor;
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::Yellow);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_title_y / dpiScalingFactor, editorsv_connectionMessageFontSize, "Multiplayer Editor", true);
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::White);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_debugtext_y / dpiScalingFactor, editorsv_connectionMessageFontSize, m_centerViewportDebugText.c_str(), true);
+        }
+
+        // Build the connection status string (just show client connected or disconnected status for now)
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();        
+        if (agentType != MultiplayerAgentType::Client)
+        {
+            return;
+        }
+
+        // Display the connection status to the top-left viewport
+        if (AzNetworking::INetworkInterface* networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName)))
```

Comment:
Nit: May worth checking whether AZ::Interface<AzNetworking::INetworking>::Get() is nullptr here.

Reply:
MP depends on AzNetworking so this _shouldn't_ be necessary. If this Get is nullptr then something has violated that dependency.